src/Symfony/Component/Serializer/Encoder/SerializerAwareEncoder.php

```diff
@@ -11,23 +11,17 @@
 
 namespace Symfony\Component\Serializer\Encoder;
 
-use Symfony\Component\Serializer\SerializerInterface;
 use Symfony\Component\Serializer\SerializerAwareInterface;
+use Symfony\Component\Serializer\SerializerAwareTrait;
 
 /**
  * SerializerAware Encoder implementation.
  *
  * @author Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * @deprecated since version 3.1, to be removed in 4.0. Use the SerializerAwareTrait instead.
```

Comment:
But that would trigger a deprecation warning whenever XmlEncoder is used, with no option to fix it.

I will update to 3.2